### PR TITLE
Add import youtube subtitles

### PR DESCRIPTION
**Pull request by kimsible**
**Issue**: https://github.com/Chocobozzz/PeerTube/issues/1016

### CODE REVIEW

File: server/helpers/youtube-dl.ts [hunk 1]
```diff
@@ -45,6 +51,35 @@ function getYoutubeDLInfo (url: string, opts?: string[]): Promise<YoutubeDLInfo>
...
+  return new Promise<YoutubeDLSubs>((res, rej) => {
+    const cwd = CONFIG.STORAGE.TMP_DIR
+    const options = opts || { all: true, format: 'vtt', cwd }
+
+    safeGetYoutubeDL()
+      .then(youtubeDL => {
+        youtubeDL.getSubs(url, options, (err, files) => {
+          if (err) return rej(err)
+
+          const subtitles = files.reduce((acc, filename) => {
```
Comment (Chocobozzz) [inline server/helpers/youtube-dl.ts:64]: Maybe a `.filter` based on the regex and a `.map` would be simpler?

**Comment by Chocobozzz** [comment]
Thanks for the PR. Please could you add unit tests?

In https://github.com/Chocobozzz/PeerTube/blob/develop/server/tests/api/videos/video-imports.ts

Run you test using `npm run mocha -- --exit -r ts-node/register -r tsconfig-paths/register --bail server/tests/your-test.ts`

I added subtitles to the youtube video we use to test youtube imports: https://www.youtube.com/watch?v=msX3jv1XdvM

File: server/helpers/youtube-dl.ts (hunk 1 above)
Comment (kimsible) [reply server/helpers/youtube-dl.ts:64]: A `.reduce` seems more appropriate to match the language code ? 
With `.filter` and `.map` we'll need two `.filter` to validate extension / language code and to match the language code.

Comment (Chocobozzz) [reply server/helpers/youtube-dl.ts:64]: oup's, you're right :+1:

**Comment by kimsible** [comment]
> Thanks for the PR. Please could you add unit tests?
> 
> In https://github.com/Chocobozzz/PeerTube/blob/develop/server/tests/api/videos/video-imports.ts
> 
> Run you test using `npm run mocha -- --exit -r ts-node/register -r tsconfig-paths/register --bail server/tests/your-test.ts`
> 
> I added subtitles to the youtube video we use to test youtube imports: https://www.youtube.com/watch?v=msX3jv1X

> npm run mocha -- --exit -r ts-node/register -r tsconfig-paths/register --bail server/tests/your-test.ts

Thanks for adding subtitles to the youtube video :)

I added some unit tests, but can't confirm it's working, mocha does not pass the first timeout and output : 

```bash
  1) Test video imports
       "before all" hook for "Should import videos on server 1":
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/kimmy/peertube-git/server/tests/api/videos/video-imports.ts)
```

These tests are passed through the CI ?

**Comment by Chocobozzz** [comment]
I'll fix the tests. But it's weird you have have this timeout issue :thinking:

**Comment by rigelk** [comment]
Thanks a bunch @kimsible - this feature will definitely prove useful to many!

**Comment by kimsible** [comment]
> I'll fix the tests. But it's weird you have have this timeout issue thinking

Thanks, I'll explore my install of PeerTube, maybe it's caused by a lack of memory.

**Pull request merged**